docs/crds/hostrule.md

```diff
@@ -309,6 +309,97 @@ The `networkSecurityPolicy` setting, in addition to any other parameters provide
 2. The HostRule CRD is not aware of the misconfigurations if it is applied to Child VS while it is being created, therefore the HostRule will be `Accepted` nonetheless. AKO will print warning message regarding this.
 
+#### Enable Regular Expression in path
+The `useRegex` field is an optional field where the user will specify if at least one path for the specified host in Kubernetes Ingress or OpenShift Route is a regular expression. This field should be set to **True** if at least one path is a regex. If the useRegex field is set to **True** for a given host, then all paths defined for that host, regardless of what Ingress/Route they are defined on, will be considered as case-insensitive regular expressions.
+
+        fqdn: test-avi.vmware.com
+        fqdnType: Exact
+        useRegex: True
+
+Avi will always follow the longest match approach to match path in the incoming request url, against the available regex paths for the host.
+
+        apiVersion: networking.k8s.io/v1
+        kind: Ingress
+        metadata:
+          name: test-ingress
+        spec:
+          rules:
+          - host: test-avi.vmware.com
+            http:
+              paths:
+              - backend:
+                  service:
+                    name: my-app-0
+                    port:
+                      number: 80
+                path: /bar.*
+                pathType: ImplementationSpecific
+              - backend:
+                  service:
+                    name: my-app-1
+                    port:
+                      number: 80
+                path: /bar/foo
+                pathType: ImplementationSpecific
+          tls:
+          - hosts:
+            - test-avi.vmware.com
+            secretName: ingress-secret
+
+For the above Ingress definition, if we create a hostrule with `useRegex` set to True, then :
+1. A request that comes on the URL **https://test-avi.vmware.com/bar/foo** will be routed to backend **my-app-1**.
+2. A request that comes on the URL **https://test-avi.vmware.com/bar/baz** will be routed to backend **my-app-0**.
+
+***Note***
+1. This property is available only in HostRule `v1beta1` schema definition.
+2. the `useRegex` field will not be supported for insecure SNI and insecure SNI dedicated virtual services.
+
+#### Express Application Root Redirect path
+The `applicationRootPath` is an optional field where the user can specify the application-root redirect path. If the application root, i.e., path `/` for a given host, is defined in a Kubernetes Ingress or OpenShift Route, but exposed on a different path in the backend and needs to be redirected, set this field to redirect requests to the required path. The host and query strings in the redirect URI will be kept the same. The port in the redirect request will be set as the front end port specified in the ingress and redirect protocol will be set based on the redirect port (HTTPS for ports 443 and 6443 and HTTP for any other port).
```

Comment:
Should we re-write this `The `applicationRootPath` is an optional field where the user can specify the application-root redirect path. If the application root, i.e., path `/` for a given host, is defined in a Kubernetes Ingress or OpenShift Route, but exposed on a different path in the backend and needs to be redirected, set this field to redirect requests to the required path` as 

`If the Application root path, defined in Ingress or Route, is exposed in a different path and needs to be redirected then set this field to the required path in a hostrule to redirect requests. This is an optional field`

Reply:
Done.